Firestore/core/src/firebase/firestore/util/executor_libdispatch.mm

```diff
@@ -246,28 +260,33 @@ void MarkDone() {
   // guaranteed to outlive the executor, and it's possible for work to be
   // invoked by libdispatch after the executor is destroyed. Executor only
   // stores an observer pointer to the operation.
+  TimeSlot* time_slot = nullptr;
+  TimeSlotId time_slot_id = 0;
+  RunSynchronized(this, [this, delay, &operation, &time_slot, &time_slot_id] {
+    time_slot_id = time_slot_counter_++;
```

Comment:
Note: overall, this approach is similar to what `ExecutorStd` does. In that case, the increment is done in a separate function, partially to make it stand out and partially to provide a natural place for the comment on wrapping: https://github.com/firebase/firebase-ios-sdk/blob/8c0ab30279d5aa347571b8a04375b9979c96633b/Firestore/core/src/firebase/firestore/util/executor_std.cc#L110-L116

Perhaps consider doing the same? It would make the two executor implementations slightly more similar, and it would formalize the comment from the PR description.

Reply:
Rephrased to mostly match.